pkg/services/queryhistory/database.go

```diff
@@ -87,9 +87,9 @@ func (s QueryHistoryService) searchQueries(ctx context.Context, user *user.Signe
 		countBuilder := db.SQLBuilder{}
 		countBuilder.Write(`SELECT
 		`)
-		writeStarredSQL(query, s.store, &countBuilder)
+		writeStarredSQL(query, s.store, &countBuilder, true)
 		writeFiltersSQL(query, user, s.store, &countBuilder)
-		err = session.SQL(countBuilder.GetSQLString(), countBuilder.GetParams()...).Find(&allQueries)
+		_, err = session.SQL(countBuilder.GetSQLString(), countBuilder.GetParams()...).Get(&totalCount)
```

Comment:
🚀 🚀 🚀

Reply:
This looks much more efficient.

Reply:
👍 Is there a way to add a test for it? Out of curiosity, did you do any benchmarks?

Reply:
@ifrost Looking at the tests in `queryhistory_search_test.go`, it already looks fairly comprehensive (testing if it wants the count aggregation or not, and if we should return only starred results or not) and will test these code changes coming from the search handler. This function is the only one that uses `writeStarredSQL`